fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -594,14 +596,77 @@ public Set<String> getUpdatedPartitionNamesOfOlapTable(OlapTable baseTable) {
         return result;
     }
 
-    private boolean isLessThanMVRewriteStaleness() {
+    /**
+     * @return Return max timestamp of all table's max refresh timestamp
+     *          which is computed by checking all its partitions' modified time.
+     */
+    public Optional<Long> maxBaseTableRefreshTimestamp() {
+        long maxRefreshTimestamp = -1;
+        for (BaseTableInfo baseTableInfo : baseTableInfos) {
+            Table baseTable = baseTableInfo.getTable();
+            if (baseTable instanceof OlapTable) {
+                OlapTable olapTable = (OlapTable) baseTable;
+                if (olapTable instanceof MaterializedView) {
```

Comment:
if (table is MV) 
if (table is Olap) 
if (table is externa) ?

Reply:
what if it's a view

Reply:
> what if it's a view
Good. Ignore view because we can get refresh timestamp from base tables. Other types' base table info will return directly.